src/depth_pro/depth_pro.py

```diff
@@ -71,8 +71,8 @@ def create_backbone_model(
 
 def create_model_and_transforms(
     config: DepthProConfig = DEFAULT_MONODEPTH_CONFIG_DICT,
-    device: torch.device = torch.device("cpu"),
-    precision: torch.dtype = torch.float32,
+    device: torch.device = torch.device("cuda" if torch.cuda.is_available() else "cpu"),
+    precision: torch.dtype = torch.float16 if device.type == 'cuda' else torch.float32,
```

Comment:
This use of `device` at this point doesn't seem to work for me.
I have to do this `precision = torch.float16 if device.type == 'cuda' else torch.float32` in the body of the function

Reply:
When running `depth-pro-run -i PATH_TO_MY_IMG_FILE`, I also get a problem:
```
ml-depth-pro/src/depth_pro/depth_pro.py:137: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
```